### PR TITLE
Preference to Handle validation disabling

### LINKED ISSUE #260: Add preference to enable/disable validation
It would be nice to have a possibility to disable XML validation

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #260

### CODE REVIEW

**Comment by NikolasKomonen** [comment]
Change from 'diagnostics' to 'validation' for the namespace.

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/settings/XMLValidationSettings.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
+package org.eclipse.lsp4xml.extensions.contentmodel.settings;
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/settings/XMLValidationSettings.java]: copyright header

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/settings/XMLProblems.java [hunk 2]
```diff
@@ -1,64 +1,64 @@
-/**
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/settings/XMLProblems.java]: please delete the file instead

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLDiagnostics.java [hunk 3]
```diff
@@ -32,14 +33,20 @@ public XMLDiagnostics(XMLExtensionsRegistry extensionsRegistry) {
...
-			doBasicDiagnostics(xmlDocument, diagnostics, monitor);
-		} catch (BadLocationException e) {
-			e.printStackTrace();
+		if(validationSettings == null || validationSettings.isEnabled()) {
+			try {
+				//Doesn't do anything ATM
+				doBasicDiagnostics(xmlDocument, diagnostics, monitor);
+				
+			} catch (BadLocationException e) {
+				e.printStackTrace();
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLDiagnostics.java]: use LOGGER instead

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/ClientSettingsManager.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+package org.eclipse.lsp4xml.settings;
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/ClientSettingsManager.java]: copyright header

File: org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java [hunk 5]
```diff
@@ -19,6 +19,7 @@
  * All defaults should be set here to eventually be overridden if needed.
  */
 public class XMLFormattingOptions extends FormattingOptions {
+
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/settings/XMLFormattingOptions.java]: unnecessary change

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLProblemsTest.java [hunk 6]
```diff
@@ -12,12 +12,14 @@
 
 import static org.eclipse.lsp4xml.XMLAssert.r;
 
+import javax.xml.crypto.dsig.XMLValidateContext;
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLProblemsTest.java]: nope

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java [hunk 7]
```diff
@@ -315,4 +323,6 @@ public void testXMLDeclUnterminated() throws Exception {
 		String xml = "<?xml version=\"1.0\" encoding=\"UTF-8\"?";
 		testDiagnosticsFor(xml, d(0, 37, 0, 37, XMLSyntaxErrorCode.XMLDeclUnterminated));
 	}
+
```
Comment (fbricon) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/extensions/contentmodel/XMLSyntaxDiagnosticsTest.java]: unnecessary
